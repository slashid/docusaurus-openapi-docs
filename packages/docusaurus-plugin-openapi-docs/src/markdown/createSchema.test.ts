/* ============================================================================
 * Copyright (c) Palo Alto Networks
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 * ========================================================================== */

import * as prettier from "prettier";

import { createNodes } from "./createSchema";
import { SchemaObject } from "../openapi/types";

describe("createNodes", () => {
  it("should create readable MODs for oneOf primitive properties", () => {
    const schema: SchemaObject = {
      type: "object",
      properties: {
        oneOfProperty: {
          oneOf: [
            {
              type: "object",
              properties: {
                noseLength: {
                  type: "number",
                },
              },
              required: ["noseLength"],
              description: "Clown's nose length",
            },
            {
              type: "array",
              items: {
                type: "string",
              },
              description: "Array of strings",
            },
            {
              type: "boolean",
            },
            {
              type: "number",
            },
            {
              type: "string",
            },
          ],
        },
      },
    };
    expect(
      createNodes(schema, "request").map((md: any) =>
        prettier.format(md, { parser: "babel" })
      )
    ).toMatchSnapshot();
  });
});
