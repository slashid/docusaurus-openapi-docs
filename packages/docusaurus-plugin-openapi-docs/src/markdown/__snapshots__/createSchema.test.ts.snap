// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`createNodes should create readable MODs for oneOf primitive properties 1`] = `
Array [
  "<SchemaItem collapsible={true} className={\\"schemaItem\\"}>
  <details style={{}} className={\\"openapi-markdown__details\\"}>
    <summary style={{}}>
      <strong>oneOfProperty</strong>
      <span style={{ opacity: \\"0.6\\" }}> object</span>
    </summary>
    <div style={{ marginLeft: \\"1rem\\" }}></div>
    <div>
      <span className={\\"badge badge--info\\"}>oneOf</span>
      <SchemaTabs>
        <TabItem label={\\"MOD1\\"} value={\\"0-item-properties\\"}>
          <SchemaItem
            collapsible={false}
            name={\\"noseLength\\"}
            required={true}
            schemaName={\\"number\\"}
            qualifierMessage={undefined}
            schema={{ type: \\"number\\" }}
          ></SchemaItem>
        </TabItem>
        <TabItem label={\\"MOD2\\"} value={\\"1-item-properties\\"}>
          <li>
            <div
              style={{
                fontSize: \\"var(--ifm-code-font-size)\\",
                opacity: \\"0.6\\",
                marginLeft: \\"-.5rem\\",
                paddingBottom: \\".5rem\\",
              }}
            >
              Array [
            </div>
          </li>
          <div style={{ marginTop: \\".5rem\\", marginBottom: \\".5rem\\" }}>
            string
          </div>
          <li>
            <div
              style={{
                fontSize: \\"var(--ifm-code-font-size)\\",
                opacity: \\"0.6\\",
                marginLeft: \\"-.5rem\\",
              }}
            >
              ]
            </div>
          </li>
        </TabItem>
        <TabItem label={\\"MOD3\\"} value={\\"2-item-properties\\"}>
          <div style={{ marginTop: \\".5rem\\", marginBottom: \\".5rem\\" }}>
            boolean
          </div>
        </TabItem>
        <TabItem label={\\"MOD4\\"} value={\\"3-item-properties\\"}>
          <div style={{ marginTop: \\".5rem\\", marginBottom: \\".5rem\\" }}>
            number
          </div>
        </TabItem>
        <TabItem label={\\"MOD5\\"} value={\\"4-item-properties\\"}>
          <div style={{ marginTop: \\".5rem\\", marginBottom: \\".5rem\\" }}>
            string
          </div>
        </TabItem>
      </SchemaTabs>
    </div>
  </details>
</SchemaItem>;
",
]
`;
